local-server/src/server/providers/ollama.ts

```diff
@@ -0,0 +1,224 @@
+import { AIProvider, AIProviderInput, AIProviderOutput, ProviderModel, ProviderConfig, ModelModality } from "./provider"
+import { APIProviderName } from "@/server/schemas/sendMessageSchema"
+import { createOllama, OllamaCompletionProviderOptions } from "ollama-ai-provider-v2"
+import { LanguageModel } from "ai"
+import { UserFacingError } from "../errors"
+import { ProviderModelFullInfo } from "./provider"
+
+type OllamaModelInfo = {
+	name: string
+	model: string
+	modified_at: string
+	size: number
+	digest: string
+	details: {
+		parent_model?: string
+		format: string
+		family: string
+		families?: string[]
+		parameter_size: string
+		quantization_level: string
+	}
+}
+
+type OllamaModelsResponse = {
+	models: OllamaModelInfo[]
+}
+
+type OllamaModelDetails = {
+	model_info: {
+		"general.architecture"?: string
+		"general.base_model.0.name"?: string
+		"general.base_model.0.organization"?: string
+		"general.parameter_count"?: number
+		"general.size_label"?: string
+		"general.license"?: string
+		[key: string]: string | number | null | undefined
+	}
+	details: {
+		parent_model: string
+		format: string
+		family: string
+		families?: string[]
+		parameter_size: string
+		quantization_level: string
+	}
+	capabilities?: string[]
+	license?: string
+	template?: string
+	system?: string
+}
+
+// Extract context length from model_info
+function extractContextLength(modelInfo: Record<string, string | number | null | undefined>): number {
+	// Search for any key ending with ".context_length"
```

Comment:
```suggestion
	// Search for any key ending with ".context_length". This is because Ollama uses scoped keys (e.g. `qwen3.context_length` / `llama.context_length`)
```

Reply:
added in 6eab8830